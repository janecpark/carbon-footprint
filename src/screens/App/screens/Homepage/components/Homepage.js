import React from 'react';
import { Link } from 'react-router-dom';
// import background from '../images/background.jpg';
// import './Homepage.css';

const Homepage = () => {
  // const style = {
  //   backgroundImage: `url(${background})`,
  //   backgroundSize: 'cover',
  //   height: '100vh',
  //   backgroundRepeat: 'no-repeat',
  //   backgroundPosition: 'center',
  //   width: '100vw',
  // };
  return (
    <div className="Homepage">
      <div className="container header">
        <h2>Welcome to Carbon Footprint Calculator</h2>
        <Link to="/household1" className="btn btn-warning">
          Take the first step
        </Link>
      </div>
    </div>
  );
};

export default Homepage;
