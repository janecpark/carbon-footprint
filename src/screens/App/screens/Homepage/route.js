import { routeConstants } from '../../../../shared/constants';
import Homepage from "./";

export default {
    path: routeConstants.HOMEPAGE.route,
    component: Homepage
};