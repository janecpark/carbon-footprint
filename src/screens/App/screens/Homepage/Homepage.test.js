import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router';
import Homepage from './components/Homepage';
import {
  UserProvider,
  FormProvider,
  ResultProvider,
  ActionProvider,
} from '../testUtils';

it('renders without crashing', function () {
  render(
    <MemoryRouter>
      <UserProvider>
        <FormProvider>
          <ResultProvider>
            <ActionProvider>
              <Homepage />
            </ActionProvider>
          </ResultProvider>
        </FormProvider>
      </UserProvider>
    </MemoryRouter>
  );
});

it('matches snapshot', function () {
  const { asFragment } = render(
    <MemoryRouter>
      <UserProvider>
        <FormProvider>
          <ResultProvider>
            <ActionProvider>
              <Homepage />
            </ActionProvider>
          </ResultProvider>
        </FormProvider>
      </UserProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

test('renders text on page', () => {
  const { getByText } = render(
    <MemoryRouter>
      <UserProvider>
        <FormProvider>
          <ResultProvider>
            <ActionProvider>
              <Homepage />
            </ActionProvider>
          </ResultProvider>
        </FormProvider>
      </UserProvider>
    </MemoryRouter>
  );
  const element = getByText(/Welcome/);
  expect(element).toBeInTheDocument();
});