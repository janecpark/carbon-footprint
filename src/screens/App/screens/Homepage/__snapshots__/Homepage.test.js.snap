// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Homepage"
    style="background-image: url(earthbg.png); background-size: auto; height: 90vh; background-repeat: no-repeat; background-position: center; width: 100vw;"
  >
    <div
      class="container header text-center "
    >
      <h2
        class="Homepage-header"
      >
        Welcome to Carbon Footprint Calculator
      </h2>
      <a
        class="btn btn-warning text-center"
        href="/household1"
        id="stepBtn"
      >
        Take the first step
      </a>
    </div>
  </div>
</DocumentFragment>
`;
